### ISSUE TITLE
false positive nonstandard_macro_braces in derive macro

### ISSUE THREAD

**Issue by caudate-julie** [rust-lang/rust-clippy#7547]
Lint name:
`nonstandard_macro_braces`

I tried this code:

```rust
// [dependencies]
// postgres-types = { version = "0.2.1", features = ["derive"] }

#[derive(postgres_types::FromSql)]
enum Example {
    Hello,
    World
}
```

I expected to see no clippy lints.

Instead, I got:

```
warning: use of irregular braces for `format!` macro
 --> src\main.rs:1:10
  |
1 | #[derive(postgres_types::FromSql)]
  |          ^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(clippy::nonstandard_macro_braces)]` on by default
help: consider writing `postgres_types::FromSql`
 --> src\main.rs:1:10
  |
1 | #[derive(postgres_types::FromSql)]
  |          ^^^^^^^^^^^^^^^^^^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#nonstandard_macro_braces
  = note: this warning originates in the derive macro `postgres_types::FromSql` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Issue was [reported in postgres](https://github.com/sfackler/rust-postgres/issues/815), but they say it's probably a bug on clippy side.

### Meta

- `cargo clippy -V`: clippy 0.1.55 (d2b04f07 2021-07-07)
- `rustc -Vv`:
```
rustc 1.55.0-nightly (d2b04f075 2021-07-07)
binary: rustc
commit-hash: d2b04f075c0ce010758c4c8674152ff89d1d73f3
commit-date: 2021-07-07
host: x86_64-pc-windows-msvc
release: 1.55.0-nightly
LLVM version: 12.0.1
```

**Comment by giraffate** [comment]
I can't reproduce it. Can you try it by using latest nightly?

It might be fixed at https://github.com/rust-lang/rust-clippy/pull/7431.

**Comment by jplatte** [comment]
I think this can be closed as fixed by #7431 given it can't be reproduced and the issue creator doesn't reply.
